src/de/simple_type.rs

```diff
@@ -493,6 +492,16 @@ impl<'de, 'a> Deref for CowRef<'de, 'a> {
     }
 }
 
+impl<'de, 'a> From<CowRef<'de, 'a>> for Content<'de, 'a> {
```

Comment:
Can one of these two structs be removed, after all of these simplifications?

Reply:
I'm not sure that this possible. `Content::Owned` stores data + offset to unconsumed data whereas `CowRef::Owned` only a content itself. `Input` and `Slice` are always store unconsumed data and updated when data is consumed